app/controllers/deploys_controller.rb

```diff
@@ -49,6 +49,87 @@ def recent
     end
   end
 
+  # little search endpoint.  
+  # takes the same tokens that are used by the filter stuff
+  # on the recent deploys pages.  returrns a paginated 
+  # json object or CSV of stuff that people are interested in rather than 
+  # doing client side filtering.
+  # params:
+  #   * username (name of the user that started the job(s)
+  #   * project_name (name of the project)
+  #   * user_type (robot or not) (not implemented)
+  #   * stage (boolean, is this in proudction or not)
+  #   * status (what is the status of this job failed|running| etc)
+  #
+  def search
+    respond_to do |format|
+
+      # stuff that we may have gotten from the api call
+      username      = (defined? params[:username]) ? params[:username] : false
+      project_name  = (defined? params[:project_name]) ? params[:project_name] : false
+      user_type     = (defined? params[:user_type]) ? params[:user_type] : false # TODO: not sure about this
+      status        = (defined? params[:status]) ? params[:status] : false
+      stage         = (defined? params[:stage]) ? params[:stage] : false
+
+      users       = false
+      projects    = false
+      jobs        = false
+      stages      = false
+
+      deploys     = { :deploys => [] }
+
+      # TODO: really need some kinda layer with error handling/messages
+      # etc
+      if (status)
+        if !Job.valid_status?(status)
+          render json: { :errors => "invalid status given" }, status: 422
+          return
+        end
+      end
+      if username
+        users = User.where("name LIKE ?", "%#{username}%").pluck(:id)
```

Comment:
escape everything that goes into like with `query = ActiveRecord::Base.send(:sanitize_sql_like, query)`